src/Accounts/Authentication/Utilities/AzConfigReader.cs

```diff
@@ -31,22 +31,50 @@ private static IAzureSession Session
             }
         }
 
+        private static IConfigManager instance = null;
+
+        public static IConfigManager Instance
+        {
+            get
+            {
+                if (instance == null)
+                {
+                    if (!Session.TryGetComponent<IConfigManager>(nameof(IConfigManager), out instance))
+                    {
+                        instance = null;
+                    }
+                }
+                return instance;
+            }
+        }
+
         public static T GetAzConfig<T>(string key, T defaultValue = default(T))
         {
-            return Session.TryGetComponent<IConfigManager>(nameof(IConfigManager), out IConfigManager configManager) ? configManager.GetConfigValue<T>(key) : defaultValue;
+            return Instance != null ? Instance.GetConfigValue<T>(key) : defaultValue;
+        }
+
+        public static void DisableWam()
+        {
+            cachedWamConfigValue = false;
+            Instance?.UpdateConfig(ConfigKeys.EnableLoginByWam, false, ConfigScope.CurrentUser);
         }
 
         static public bool IsWamEnabled(string authority)
         {
-            if (!string.IsNullOrEmpty(authority) && Session.TryGetComponent<IConfigManager>(nameof(IConfigManager), out var config))
+            if (cachedWamConfigValue != null)
```

Comment:
Do not cache config values - they are already cached in config manager, and we want to keep the log and telemetry (in the future) of accessing configs.